### ISSUE TITLE
mx.model.FeedForward.predict has something wrong

### ISSUE THREAD

**Issue by qinhui99** [apache/incubator-mxnet#5662]
For bugs or installation issues, please provide the following information.
The more information you provide, the more likely people will be able to help you.

## Environment info
Operating System:

Compiler:

Package used (Python/R/Scala/Julia):

MXNet version:0.93,0.94

Or if installed from source:

MXNet commit hash (`git rev-parse HEAD`):

If you are using python package, please provide

Python version and distribution:

If you are using R package, please provide

R `sessionInfo()`:

## Error Message:
Please paste the full error message, including stack trace.
I use iris dataset to do some tests.I build some neural network based on mxnet. After training ,I do some prediction with the test dataset, it seems I get the best good performance : 
"'ce', 'acc'", [0.0025108862668275834, 1.0].
But I compare the original labels and  the prediction, only 50% of them is right. 

And I use the trained model to predict some new samples, the prediction are also very bad.  I am confused for these result. So I made some codes in Tensorflow to do the same things. The Tensorflow codes works very well. 

## Minimum reproducible example
if you are using your own code, please provide a short script that reproduces the error.
#encoding=utf-8
import mxnet as mx

import logging
import numpy as np

mx.random.seed(1)
# mlp
net = mx.sym.Variable('data')
net = mx.sym.FullyConnected(net, name='fc1', num_hidden=10)
net = mx.sym.Activation(net, name='relu1', act_type="relu")
net = mx.sym.FullyConnected(net, name='fc2', num_hidden=3)
net = mx.sym.SoftmaxOutput(net, name='softmax')

# We use utils function in sklearn to get iris dataset in pickle
from sklearn import datasets
from sklearn.utils import shuffle
from sklearn.preprocessing import StandardScaler
iris = datasets.load_iris()

# Normalize data
scalar=StandardScaler()
scalared_data=scalar.fit_transform(iris.data)
# shuffle data
X, y = shuffle(scalared_data, iris.target)
# X, y = (scalared_data, iris.target)

split_index=90
split_index2=100
batch_size = split_index2-split_index

train_data =np.concatenate((X[:split_index],X[split_index2:]),axis=0).astype('float32')
train_label = np.append(y[:split_index],y[split_index2:])

val_data = X[split_index :split_index2].astype('float32')
val_label = y[split_index:split_index2]

# Build iterator
train_iter = mx.io.NDArrayIter(data=train_data, label=train_label, batch_size=batch_size, shuffle=True)
val_iter = mx.io.NDArrayIter(data=val_data, label=val_label, batch_size=batch_size, shuffle=True)
logging.basicConfig(level=logging.INFO)
epoch_num=20
lr=0.07
devs = mx.cpu()
mod = mx.model.FeedForward(
    ctx=devs,
    symbol=net,
    num_epoch=epoch_num,
    learning_rate=lr,
    momentum=0.9,
    optimizer='sgd')

mod.fit(X = train_iter, eval_metric = 'acc',  eval_data=val_iter)
print('Finished training')

# import time
# time.sleep(1)
y = mod.predict(val_iter,num_batch=1,reset=False)
#The evaluation labels :2 1 1 0 0 1 0 0 0 2
for b in val_label[:batch_size]:
    print (b),
print

#The prediction labels:0 1 2 0 1 0 0 0 1 2 . Only 50% of them is right.
for a in y :
    max=np.argmax(a)
    print (max),

print
#But the score is ("'ce', 'acc'", [0.0025108862668275834, 1.0]) . What's happen?
print ("'ce', 'acc'",mod.score(val_iter, ['ce', 'acc'],num_batch=1))
# mod.save("iris",epoch=epoch_num)

def get_predict_result(a):

    result = mod.predict(mx.io.NDArrayIter(data=a, label=None, batch_size=1))
    print('shape of predict: %s' % (result.shape,), result)
    for a in result:
        max = np.argmax(a)
        print (max),

#The lables should be [1,0,2,1]
new_samples = np.array(
        [[5.5, 2.3, 4.0, 1.3], [5.0, 3.3, 1.4, 0.2], [5.8, 2.7, 5.1, 1.9], [5.5, 2.6, 4.4, 1.2]], dtype=float)
#My Model seems have a very good result :("'ce', 'acc'", [0.0024675801396369934, 1.0]). But it is not true.
# The lables should be [1,0,2,1],But I got the prediction labels as [2,1,2,2]
get_predict_result(new_samples)

## Steps to reproduce
or if you are running standard examples, please provide the commands you have run that lead to the error.

1.  Run the codes, you can see the printing information. 
2.
3.

## What have you tried to solve it?

1. 

2.
3.

**Comment by szha** [comment]
This issue is closed due to lack of activity in the last 90 days. Feel free to ping me to reopen if this is still an active issue. Thanks!
